### ISSUE TITLE
Request URL containing $ doesn't work well with PHP - cURL

### ISSUE THREAD

**Issue by honeyyadav** [postmanlabs/postman-code-generators#284]
# Short Explanation

When I am generating **PHP - cURL** code from a postman request with URL containing a **$** sign followed by a **letter** or **underscore** (for example $firstName, $_address, $LastName), then the code generated by postman doesn't work. This is because PHP treats them as variable name (but they are not variable name(s) in this case).

# Example

My request URL is : -
```
> http://example.com/path?$filter=name eq John
```
The code generated by postman : -
```php
<?php

$curl = curl_init();

curl_setopt_array($curl, array(
  CURLOPT_URL => "http://example.com/path?$filter=name%20eq%20John",
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => "",
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => "GET",
));

$response = curl_exec($curl);

curl_close($curl);
echo $response;

```

The above code will not work as PHP will treat **\$filter** as a variable name. To solve this problem we need to put a **\\** (backslash) before **\$filter** like this : -

```php
<?php

$curl = curl_init();

curl_setopt_array($curl, array(
  CURLOPT_URL => "http://example.com/path?\$filter=name%20eq%20John",
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => "",
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => "GET",
));

$response = curl_exec($curl);

curl_close($curl);
echo $response;

```

And now it will work **flawlessly**.

> **Postman version used:** v7.26.0

**Comment by erdemkose** [comment]
Instead of escaping `$` characters, replacing double quotes(`"`) with single quotes(`'`) would be a simpler and cleaner solution. PHP does not look for variables in single-quoted strings.

**Comment by shellstrom** [comment]
I'm not entirely sure this is the exact same thing, but I'm errors from an API when I use the PHP - cURL option and the postfields look like this. I can run this no problem with bash curl export equivalent in Postman, but for some reason this isn't working out of the box with the PHP - cURL option.
However, @honeyyadav 's suggestion worked out for me, by prepending every `$` with a `\`, resulting in `\$`...
```
CURLOPT_POSTFIELDS` =>"[\r\n    {\r\n        \"name\": null,\r\n        \"variables\": {\r\n            \"postId\": \"502\",\r\n            \"next\": \"520\",\r\n            \"previous\": null,\r\            \"start\": 50,\r\n            \"end\": null\r\n        },\r\n        \"query\": \"query ($postId: ID!, $next: String, $previous: String, $start: Int, $end: Int) {\\n  posts: article(id: $postId) {...
```

**Comment by webholik** [comment]
Fixed by #352
